src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -13979,5 +13948,278 @@ int32[] arg2
+                int MethodNoParams(int[] xs) => xs.Length;
+                int MethodWithParams(params int[] xs) => xs.Length;
+                DelegateNoParams dNoParams;
+                DelegateWithParams dWithParams;
+                dNoParams = (params int[] xs) => xs.Length; // 1
+                dNoParams = (int[] xs) => xs.Length;
+                dNoParams = MethodNoParams;
+                dNoParams = MethodWithParams;
+                dWithParams = (params int[] xs) => xs.Length;
+                dWithParams = (int[] xs) => xs.Length;
+                dWithParams = MethodNoParams;
+                dWithParams = MethodWithParams;
+                delegate int DelegateNoParams(int[] xs);
+                delegate int DelegateWithParams(params int[] xs);
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (5,27): warning CS9502: Parameter 1 has params modifier in lambda but not in target delegate type.
+                // dNoParams = (params int[] xs) => xs.Length; // 1
+                Diagnostic(ErrorCode.WRN_ParamsArrayInLambdaOnly, "xs").WithArguments("1").WithLocation(5, 27));
+        }
+
+        [Fact]
+        public void ParamsArray_CommonType()
+        {
+            var source = """
+                int MethodNoParams(int[] xs) => xs.Length;
+                int MethodWithParams(params int[] xs) => xs.Length;
+                var inferredNoParams = MethodNoParams;
+                var inferredWithParams = MethodWithParams;
+                var lambdaNoParams = (params int[] xs) => xs.Length;
+                var lambdaWithParams = (int[] xs) => xs.Length;
+                var a1 = new[] { MethodNoParams, MethodNoParams };
+                var a2 = new[] { inferredNoParams, inferredWithParams }; // 1
+                var a3 = new[] { lambdaNoParams, lambdaWithParams }; // 2
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (8,10): error CS0826: No best type found for implicitly-typed array
+                // var a2 = new[] { inferredNoParams, inferredWithParams }; // 1
+                Diagnostic(ErrorCode.ERR_ImplicitlyTypedArrayNoBestType, "new[] { inferredNoParams, inferredWithParams }").WithLocation(8, 10),
+                // (9,10): error CS0826: No best type found for implicitly-typed array
+                // var a3 = new[] { lambdaNoParams, lambdaWithParams }; // 2
+                Diagnostic(ErrorCode.ERR_ImplicitlyTypedArrayNoBestType, "new[] { lambdaNoParams, lambdaWithParams }").WithLocation(9, 10));
+        }
+
+        [Fact]
+        public void ParamsArray_CommonType_Methods()
+        {
+            var source = """
+                int MethodNoParams(int[] xs) => xs.Length;
+                int MethodWithParams(params int[] xs) => xs.Length;
+                var a = new[] { MethodNoParams, MethodNoParams };
+                System.Console.WriteLine(a.GetType());
+                """;
```

Comment:
>"""; [](http://example.com/codeflow?start=16&length=4)

Consider adding:
```
var a2 = new[] { (int[] xs) => xs.Length, (params int[] xs) => xs.Length };
System.Console.WriteLine(a2.GetType());
```

Reply:
This would fail and is already tested above as you suggested in https://github.com/dotnet/roslyn/pull/64961#discussion_r1014231281.

Actually, this whole test was wrong as `MethodWithParams` wasn't used. Removed it: 7effba29e6ddc7aef97e2f78c0ec1f6f39870eb9